app/models/SupplementaryRequest.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import cats.data.EitherNec
+import cats.implicits.catsSyntaxTuple3Parallel
+import pages.{HasSupplementaryUnitUpdatePage, SupplementaryUnitUpdatePage}
+import play.api.libs.json.{Json, OFormat}
+import queries.MeasurementQuery
+
+final case class SupplementaryRequest(
+  eori: String,
+  recordId: String,
+  hasSupplementaryUnit: Option[Boolean],
+  supplementaryUnit: Option[String],
+  measurementUnit: Option[String]
+)
+
+object SupplementaryRequest {
+
+  implicit lazy val format: OFormat[SupplementaryRequest] = Json.format
+
+  def build(answers: UserAnswers, eori: String, recordId: String): EitherNec[ValidationError, SupplementaryRequest] =
+    (
+      getHasSupplementaryUnit(answers, recordId),
```

Comment:
All three of these things just return Rights... so is this really an EitherNec?

Reply:
so the getPageValue actually does that, and I kept it like that here, get(page).map(Right(_)).getOrElse(Left(NonEmptyChain.one(PageMissing(page)))). Without wrapping it in Right - it was complaining so just left it :D. It works though Any suggestions?

Reply:
Ah ok I think I see what happened. Will ping you